### ISSUE TITLE
CDK Overlay: ViewportMargin per side

### ISSUE THREAD

**Issue by seritools** [angular/material2#14573]
#### Please describe the feature you would like to request.

It would be nice to be able to define the viewport margin for each side independently (similar to a css margin).

Something like

```ts
interface ViewportMargin {
    top: number,
    right: number,
    bottom: number,
    left: number
}

viewPortMargin: number | ViewportMargin;
```

would be nice to have.

#### What is the use-case or motivation for this proposal?

I'd like to make sure my overlay is only shown within the "main" area of my application and not on top of the action bar or sidenav. (Injecting a custom OverlayContainer and a second Overlay service instance limited to my desired area doesn't work since it still uses the ViewportRuler for calculation).

**Comment by lukasholzer** [comment]
Are you willing to accept a PR for this – I'm in dire need of this feature as well.

Regards

**Comment by crisbeto** [comment]
Yes, we're accepting PRs, but you should keep in mind that we usually merge in new features in minor or major releases so it might take a while for it to get released.

**Comment by crisbeto** [comment]
Should be addressed by #29563.
